### ISSUE TITLE
Pyright does not catch SyntaxError: unterminated f-string literal

### ISSUE THREAD

**Issue by namanarora42** [microsoft/pyright#9789]
<h3> Environment data </h3>

-   Language Server version: 2024.11.1
-   OS and version: win32 x64
-   Python version (and distribution if applicable, e.g. Anaconda): 3.12.7
-   python.analysis.indexing: true
-   python.analysis.typeCheckingMode: off

<h3> Code Snippet </h3>

```python
a = (f'hello
     there')

print(a)
```

<h3> Repro Steps </h3>

1. Type in the code snippet as-is into an empty .py file. 
2. Observe the "Problems" window for warnings/issues raised

<h3> Expected behavior </h3>

Pylance should highlight the syntax error in the first line. When you run this script, Python throws the below error:

```
SyntaxError: unterminated f-string literal (detected at line 1)
```

Pylance should be able to catch this error. For example, I tried typing the same snippet into PyCharm Community Edition, and it correctly identifies the errors (see screenshot)

![Correct Behavior Example from PyCharm](https://i.imgur.com/vxE40J3.png)

<h3> Actual behavior </h3>

Pylance does not report any problems with this code.

<h3> Logs </h3>

```
2025-01-29 15:12:34.565 [info] Send text to terminal: C:/Users/XXX/AppData/Local/Programs/Python/Python312/python.exe d:/Projects/XXX/Sandbox/test.py
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release of pyright.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.394.
